include/common/mem.h

```diff
@@ -0,0 +1,48 @@
+/* SPDX-License-Identifier: GPL-2.0-only */
+#ifndef __LABWC_MEM_H
+#define __LABWC_MEM_H
+
+#include <stdlib.h>
+
+/*
+ * Like calloc(), but calls exit() on error.
+ * Returns NULL only if (n * size == 0).
+ */
+void *xcalloc(size_t n, size_t size);
+
+/* Shorthand for the common case */
+#define MAKE_NEW(type, name) type *name = xcalloc(1, sizeof(type))
+
+/*
+ * Like realloc(), but calls exit() on error.
+ * Returns NULL only if (size == 0).
+ */
+void *xrealloc(void *ptr, size_t size);
+
+/* malloc() is a specific case of realloc() */
+#define xmalloc(size) xrealloc(NULL, (size))
+
+/*
+ * Like strdup(), but calls exit() on error.
+ * Returns NULL only if (str == NULL).
+ */
+char *xstrdup(const char *str);
+
+/*
+ * Frees <str> and replaces it with a copy of <new_str>.
+ * Sets <str> to NULL only if (new_str == NULL).
+ * Calls exit() on error.
+ */
+#define xreplace_str(str, new_str) do { \
+        free(str); (str) = xstrdup(new_str); \
+} while(0)
```

Comment:
I struggled to come up with a good name for `xreplace_str`. Other ideas I had were `xstrdup_to` or `xstrdup_over`.